trip_app/trips/views.py

```diff
@@ -43,3 +53,30 @@ def get_queryset(self):
                 order_by(F('dist1') + F('dist2'))
 
         return queryset
+
+    @action(detail=True, methods=['post'])
+    def reserve(self, request, pk=None):
+        """ Reserve a trip by sending a POST to api/trip/<trip_id>/reserve/.
+            If the man_approve field of the trip model is True the trip request will be created,
+            which means that user requested a trip should be approved by the driver.
+            If man_approve is False user will be added to the passengers list if there are free seats """
+        trip = self.get_object()
+        user = request.user
+        if user == trip.driver:
+            return Response(data="Driver can't be a passenger at the same time", status=status.HTTP_400_BAD_REQUEST)
+        if trip.free_seats:
+            if trip.man_approve:
+                trip_request = TripRequest.objects.create(trip=trip, user=user)
+                serializer = TripRequestSerializer(trip_request)
+                return Response(serializer.data)
```

Comment:
Why we need data on response, maybe just 201 code?

Reply:
Hm. I'm not sure what is the best practice for such cases, I used to think that it is recommended to return the entity in the response body when smth is created on the server, though indeed the response 201 should be more appropriate then 200

Reply:
any final choice here? @aleksandr-hilko

Reply:
I switched here to 201 + entity in response.